CODE_COMMENT_STYLE.md

```diff
@@ -0,0 +1,116 @@
+# Code Comment Style
+
+This document describes the code comment style applied to TiKV repositories. When you are to commit, be sure to follow the style to write good code comments.
+
+## Why does a good comment matter?
+
+- To speed up the reviewing process
+- To help maintain the code
+- To improve the API document readability
+- To improve the development efficiency of the whole team
+
+## Where/When to comment?
+
+Write a comment where/when there is context that is missing from the code or is hard to deduce from the code. Specifically, use a comment:
+
+- For important code
+- For obscure code
+- For tricky or interesting code
+- For a complex code block
+- If a bug exists in the code but you cannot fix it or you just want to ignore it for the moment
+- If the code is not optimal but you don’t have a smarter way now
+- To remind yourself or others of missing functionality or upcoming requirements not present in the code
+
+A comment is generally required for:
+
+- Module
+- Type
+- Constant
+- Function
+- Method
+- Variable
+- Typical algorithm
+- Exported name
+- Test case
+- TODO
+- FIXME

```

Comment:
Some of thee seem too demanding to me, and I might separate this into "comments" and "doc comments"; with doc comments required for: _public_ modules, types, constants, functions, methods, fields (which could be summarized a "all public items (such as modules, types, constants, functions, and fields) that are exported from the crate". That's a pretty typical Rust policy.

Then the remaining are sort of PingCAP policies, some of which I'd take issue with. For example, I would not put a comment on most variables, preferring the variable name be descriptive enough to understand its purpose from the code. It's also not clear to me what "typical algorithm" means, though I might interpret this as just "algorithms", or maybe "Identify the name of known algorithms", and "describe complex algorithms".

I'm not sure about the "exported name" case. Is this a rustdoc thing, or a tikv-thing? (I mostly don't comment `pub use`. Though maybe this just refers to any "thing" that's exported? It could be clearer.

Please give a thought to the above and see if things can be clarified a bit. I'm happy to offer suggestions if needed.

Reply:
Great inputs. 
For typical rust items, will add a general description as suggested
For common items that look confusing:

- Will not use "required" since it's denoting mandatory instead of necessary.  So no pressure.
-  Removed "exported name" since its a golang thing.
- "complex algorithm" instead of "typical